src/main/java/org/assertj/core/api/Assumptions.java

```diff
@@ -0,0 +1,757 @@
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  @SuppressWarnings("unchecked")
+  public static <T> AbstractObjectArrayAssert<?, T> assumeThat(T[] actual) {
+    return enhanceAsAssumption(ObjectArrayAssert.class, Object[].class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link MapAssert}</code> assumption.
+   *
+   * @param <K> the type of keys in the map.
+   * @param <V> the type of values in the map.
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  @SuppressWarnings("unchecked")
+  public static <K, V> AbstractMapAssert<?, ?, K, V> assumeThat(Map<K, V> actual) {
+    return enhanceAsAssumption(MapAssert.class, Map.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link ThrowableAssert}</code> assumption.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assumeThat(Throwable actual) {
+    return enhanceAsAssumption(ThrowableAssert.class, Throwable.class, actual);
+  }
+
+  /**
+   * Allows to capture and then assume on a {@link Throwable} (done easier when used with Java 8 lambdas).
+   *
+   * @param shouldRaiseThrowable The {@link ThrowingCallable} or lambda with the code that should raise the throwable.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assumeThatThrownBy(ThrowingCallable shouldRaiseThrowable) {
+    return enhanceAsAssumption(ThrowableAssert.class, Throwable.class, catchThrowable(shouldRaiseThrowable));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <ASSERTION, ACTUAL> ASSERTION enhanceAsAssumption(Class<ASSERTION> assertionType,
+                                                                   Class<ACTUAL> actualType,
+                                                                   Object actual) {
+    Enhancer enhancer = new Enhancer();
+    enhancer.setSuperclass(assertionType);
+    enhancer.setCallback(new MethodInterceptor() {
+      @Override
+      public Object intercept(Object o, Method method, Object[] args, MethodProxy methodProxy) throws Throwable {
+        try {
+          Object result = methodProxy.invokeSuper(o, args);
+          if (result != o && result instanceof AbstractAssert) {
+            return enhanceAsAssumption(result);
+          }
+          return result;
+        } catch (AssertionError e) {
+          throw createAssumptionViolatedException(e);
+        }
+      }
+    });
+
+    return (ASSERTION) enhancer.create(new Class[] { actualType }, new Object[] { actual });
+  }
+
+  private static RuntimeException createAssumptionViolatedException(AssertionError e) throws ReflectiveOperationException {
```

Comment:
would be good to find a more accurate name as we can build a `SkipException`, maybe `failingAssumption` or `unmetAssumption` (not sure this is correct english) or `assumptionNotMet`, I'm not ending with `Exception` because it is clear it is one since it comes after `throw `, I think `throw assumptionNotMet(e);` reads well, thoughts ?